Viktor_Miroshnychenko/3/poker.py

```diff
@@ -0,0 +1,244 @@
+
+        if counter == 5:
+            straight = True
+            break
+
+    return straight
+
+
+def check_straight_flash(hand):
+    """
+    :param hand: hand to check on straight flash
+    :type hand: str
+
+    :return: True(if straight flash) or False(otherwise) 
+    :rtype: bool
+    """
+
+    str_flash = check_flash(hand)
+    ret_val = str_flash
+    if str_flash:
+        ret_val = check_straight(hand)
+
+    return ret_val
+
+
+def check_four_kind(hand):
+    """
+    :param hand: hand to check on four of a kind
+    :type hand: str
+
+    :return: True if four of a kind, False otherwise
+    :rtype: bool
+    """
+
+    ret_val = False
+    hand = hand.replace(" ", "")
+    for it in range(5):
+        if hand.count(hand[it * 2]) == 4:
+            ret_val = True
+            break
+
+    return ret_val
+
+
+def check_pair(hand):
+    """
+    :param hand: hand to check on pair
+    :type hand: str
+
+    :return: True if pair present, False otherwise
+    :rtype: bool
+    """
+
+    ret_val = False
+    hand = hand.replace(" ", "")
+    for it in range(5):
+        if hand.count(hand[it*2]) == 2:
+            ret_val = True
+            break
+
+    return ret_val
+
+
+def check_three_kind(hand):
+    """
+    :param hand: hand to check on three of a kind
+    :type hand: str
+
+    :return: True if three of a kind, False otherwise
+    :rtype: bool
+    """
+
+    ret_val = False
+    hand = hand.replace(" ", "")
+    for it in range(5):
+        if hand.count(hand[it * 2]) == 3:
+            ret_val = True
+            break
+
+    return ret_val
+
+
+def check_two_pairs(hand):
+    """
+    :param hand: hand to check on two pairs
+    :type hand: str
+
+    :return: True if has three pairs, False otherwise
+    :rtype: bool
+    """
+
+    pair_amount = 0
+    hand = hand.replace(" ", "")
+    for it in range(5):
+        if hand.count(hand[it*2]) == 2:
+            pair_amount += 1
+
+    return pair_amount == 4
+
+
+def check_full_house(hand):
+    """
+    :param hand: hand to check on full house
+    :type hand: str
+
+    :return: True if full house, False otherwise
+    :rtype: bool
+    """
+
+    return check_pair(hand) and check_three_kind(hand)
+
+
+def find_highest_card(hand):
+    """
+    :param hand: hand to search a card in
+    :type hand: str
+
+    :return: highest card index in CARD_HIGH
+    :rtype: int
+    """
+
+    global cards
+    ret_val = 0
+    tmp_list = hand.split(" ")
+    tmp = ""
+    for l in tmp_list:
+        l = l[:-1]
+        tmp += l
+    hand = tmp    
+
+    for card in cards:
+        if hand.rfind(card) != -1:
+            ret_val = CARD_HIGH.index(card)
+            break
+
+    return ret_val
+
+
+def find_hand(hands):
+    """
+    :param hands: list of hands
+    :type hands: list[str]
+
+    :return: highest hand
+    :rtype: str
+    """
+
+    power_list = []
+    for index in range(len(hands)):
+         power_list.append(0)
+    for index in range(len(hands)):
```

Comment:
There's much better way to do this.

Reply:
do you mean the whole task or that particular function ?